buyer-api/src/routes/dataOrders/addDataResponse.js

```diff
@@ -0,0 +1,91 @@
+import express from 'express';
+import fetchDataOrder from './middlewares/fetchDataOrder';
+import { asyncError } from '../../utils';
+import { addDataResponse } from '../../operations/addDataResponse';
+
+const router = express.Router();
+
+/**
+ * @swagger
+ * /orders/{id}/data-responses:
+ *   post:
+ *     description: |
+ *       # STEP 4 from Wibson's Protocol
+ *       ## Buyer accumulates DataResponses to notarize them by batches.
+ *     parameters:
+ *       - in: body
+ *         name: dataOrder
+ *         type: object
+ *         required: true
+ *         schema:
+ *           $ref: "#/definitions/DataResponse"
+ *     produces:
+ *       - application/json
+ *     responses:
+ *       200:
+ *         description: When the DataResponse was accepted
+ *       422:
+ *         description: When the DataResponse has missing or invalid fields
+ *       500:
+ *         description: When there is a problema processing the DataResponse
+ *
+ * definitions:
+ *   DataResponse:
+ *     type: object
+ *     properties:
+ *       orderId:
+ *         type: number
+ *         required: true
+ *         description: Order ID in the DataExchange contract
+ *         example: '42'
+ *       sellerAddress:
+ *         type: string
+ *         required: true
+ *         description: Seller's Ethereum address
+ *         example: '"0xa42df59C5e17df255CaDfF9F52a004221f774f36"'
+ *       sellerId:
+ *         type: number
+ *         description: Seller's ID in the BatPay contract
+ *         example: '1085'
+ *       encryptedData:
+ *         type: string
+ *         required: true
+ *         description: Data encrypted with symmetric-key algorithm
+ *         example: '"tZ4MsEnfbcDOwqau68aOrQ=="'
+ *       decryptedDataHash:
+ *         type: string
+ *         required: true
+ *         description: Hash of the raw data
+ *         example: '"8f54f1c2d0eb5771cd5bf67a6689fcd6eed9444d91a39e5ef32a9b4ae5ca14ff"'
+ *       decryptionKeyHash:
+ *         type: string
+ *         required: true
+ *         description: Hash of the key used to encrypt the data
+ *         example: '"07855b46a623a8ecabac76ed697aa4e13631e3b6718c8a0d342860c13c30d2fc"'
+ *       notaryAddress:
+ *         type: string
+ *         required: true
+ *         description: Notary's Ethereum address
+ *         example: '"0xccCF90140Fcc2d260186637D59F541E94Ff9288f"'
+ *       notaryUrl:
+ *         type: string
+ *         required: true
+ *         description: Notary's service url
+ *         example: '"https://api.notary.com/notarizations"'
+ *       needsRegistration:
+ *         type: boolean
+ *         required: true
+ *         description: Whether the Seller needs to be registered in BatPay or not
+ *         example: 'true'
+ */
+router.post('/:id/data-responses', fetchDataOrder, asyncError(async (req, res) => {
```

Comment:
Absolutely agree, we have the performance in mind while building v2:
* There is no fallback requests to the blockchain to fetch the order, that is not necessary anymore.
* It is not implemented yet, but the input will be validated in a middleware (using swagger schema) returning a 422 just before the `fetchDataOrder`, so it will not access the db if there is a problem with the request.
* GET handlers will also add the headers so that clients can cache the response